File: lua/difference-of-squares/difference-of-squares.lua
local function square_of_sum(n)
  local val = 0
  for i=1, n do
    val = val + i
  end
  return val*val
end

local function sum_of_squares(n)
  local val = 0
  for i=1, n do
    val = val + i*i
  end
  return val
end

local function difference_of_squares(n)
  return square_of_sum(n) - sum_of_squares(n)
end

return {
  square_of_sum = square_of_sum,
  sum_of_squares = sum_of_squares,
  difference_of_squares = difference_of_squares
}

File: ocaml/difference-of-squares/difference_of_squares.ml
let sum l = List.fold_left (+.) 0. l

let square_of_sum n =
  let s = sum (List.init n (fun x -> float_of_int (x + 1))) in
  int_of_float (s *. s)

let sum_of_squares n =
  int_of_float (sum (List.init n (fun x -> float_of_int ((x + 1) * (x + 1)))))

let difference_of_squares n =
  square_of_sum n - sum_of_squares n
